### PR TITLE
New Unit Tests

### PR DESCRIPTION
**Pull request by tyler-travis**
@blaine-O  Please review and merge.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -40,4 +40,11 @@ client/openstorefront/app/images/logo/DI2E-(logo).gif
 /server/openstorefront/openstorefront-web/src/main/webapp/submission.html
 /server/openstorefront/apiclient/target/
 /server/openstorefront/core/target/
-/server/openstorefront/core-modules/core/target/
\ No newline at end of file
+/server/openstorefront/core-modules/core/target/
+client/openstorefront/nbproject/project.properties.BACKUP.3328.properties
```
Comment (blaine-O) [inline .gitignore]: @Jedwondle  Is it OK to add these here?

File: client/openstorefront/nbproject/project.properties [hunk 2]
```diff
@@ -7,14 +7,17 @@ auxiliary.org-netbeans-modules-css-prep.sass_2e_enabled=true
 auxiliary.org-netbeans-modules-css-prep.sass_2e_mappings=/styles:/styles
 auxiliary.org-netbeans-modules-javascript-nodejs.enabled=true
 auxiliary.org-netbeans-modules-javascript-nodejs.run_2e_enabled=true
+auxiliary.org-netbeans-modules-javascript-nodejs.run_2e_restart=false
```
Comment (blaine-O) [inline client/openstorefront/nbproject/project.properties]: @tyler-travis what does this do?

File: client/openstorefront/package.json [hunk 3]
```diff
@@ -14,7 +14,7 @@
     "grunt-bower-install": "~1.0.0",
     "grunt-concurrent": "~0.5.0",
     "grunt-contrib-clean": "~0.5.0",
-    "grunt-contrib-compass": "~0.7.2",
+    "grunt-contrib-compass": "^1.0.3",
```
Comment (blaine-O) [inline client/openstorefront/package.json]: @Jedwondle do you know any issues of upgrading compass from 0.7.2 to 1.0.3?

File: client/openstorefront/test/spec/controllers/admin.js [hunk 4]
```diff
@@ -25,17 +25,33 @@
    expect(scope.menuPanel.data.open).toEqual(true);
    expect(scope.menuPanel.system.open).toEqual(true);
    expect(scope.oneAtATime).toEqual(false);
+
```
Comment (blaine-O) [inline client/openstorefront/test/spec/controllers/admin.js]: @tyler-travis These two lines can be taken out, or I can do it in a cleanup pull request after this merges

File: client/openstorefront/test/spec/controllers/nav.js [hunk 5]
```diff
@@ -8,9 +8,10 @@
    var NavCtrl,
      scope;
 
-// Initialize the controller and a mock scope
+   // Initialize the controller and a mock scope
    beforeEach(inject(function ($controller, $rootScope) {
      scope = $rootScope.$new();
+     $rootScope.searchKey = 'test'
```
Comment (blaine-O) [inline client/openstorefront/test/spec/controllers/nav.js]: @tyler-travis @Jedwondle Do we want to set this here?  Shouldn't we let this get set by the controller?

File: client/openstorefront/test/spec/controllers/userprofile.js [hunk 6]
```diff
@@ -17,19 +17,22 @@ describe('Controller: UserProfileCtrl', function () {
   }));
 
   it('should have the correct initializations', function () {
-    expect(scope._scopename).toEqual('userprofile');
-    expect(scope.pageTitle).toEqual('DI2E Clearinghouse');
+    expect(scope.total).toEqual({});
+    expect(scope.userProfileForm).toEqual({});
+    expect(scope._scopename).toEqual('userprofile');  //NOTE this is set to 'userprofile' when scoped!
```
Comment (blaine-O) [inline client/openstorefront/test/spec/controllers/userprofile.js]: @tyler-travis The comments here can be removed; or I can remove them in a subsequent pull request.

File: client/openstorefront/test/spec/controllers/nav.js (hunk 5 above)
Comment (tyler-travis) [inline client/openstorefront/test/spec/controllers/nav.js]: @blaine-O The reason I put that there is it is similar to the way main.js is done:
```javascript
 beforeEach(inject(function ($controller, $rootScope) {
    scope = $rootScope.$new();
    $rootScope.searchKey = 'test';
    $rootScope.openAdminMessage = 'test';
    MainCtrl = $controller('MainCtrl', {
      $scope: scope
    });
```

File: client/openstorefront/nbproject/project.properties (hunk 2 above)
Comment (tyler-travis) [inline client/openstorefront/nbproject/project.properties]: @blaine-O To be honest, I don't know.

File: client/openstorefront/test/spec/controllers/nav.js (hunk 5 above)
Comment (blaine-O) [inline client/openstorefront/test/spec/controllers/nav.js]: Sounds good

**Pull request merged**